src/core/Akka.Remote.Tests/RemotingSpec.cs

```diff
@@ -161,12 +158,77 @@ public void Remoting_must_support_remote_lookups()
             //TODO: using smaller numbers for the cancellation here causes a bug.
             //the remoting layer uses some "initialdelay task.delay" for 4 seconds.
             //so the token is cancelled before the delay completed.. 
-            var msg = await here.Ask<Tuple<string,IActorRef>>("ping", TimeSpan.FromSeconds(1.5));
+            var msg = await here.Ask<Tuple<string, IActorRef>>("ping", TimeSpan.FromSeconds(1.5));
             Assert.Equal("pong", msg.Item1);
             Assert.IsType<FutureActorRef>(msg.Item2);
         }
 
         [Fact]
+        public void Remoting_must_not_send_remote_recreated_actor_with_same_name()
+        {
+            var echo = remoteSystem.ActorOf(Props.Create(() => new Echo1()), "otherEcho1");
+            echo.Tell(71);
+            ExpectMsg(71);
+            echo.Tell(PoisonPill.Instance);
+            ExpectMsg("postStop");
+            echo.Tell(72);
+            ExpectNoMsg(TimeSpan.FromSeconds(1));
+
+            var echo2 = remoteSystem.ActorOf(Props.Create(() => new Echo1()), "otherEcho1");
+            echo2.Tell(73);
+            ExpectMsg(73);
+
+            // msg to old IActorRef (different UID) should not get through
+            echo2.Path.Uid.ShouldNotBe(echo.Path.Uid);
+            echo.Tell(74);
+            ExpectNoMsg(TimeSpan.FromSeconds(1));
+
+            remoteSystem.ActorSelection("/user/otherEcho1").Tell(75);
+            ExpectMsg(75);
+
+            Sys.ActorSelection("akka.test://remote-sys@localhost:12346/user/otherEcho1").Tell(76);
+            ExpectMsg(76);
+        }
+
+        [Fact]
+        public void Remoting_must_lookup_actors_across_node_boundaries()
```

Comment:
Does a bunch of crazy cross-boundary remote deployments and verifies that our ability to resolve and select those actors still works back and forth over the network.